aiohttp/helpers.py

```diff
@@ -95,21 +95,30 @@ def __new__(cls, login: str,
     @classmethod
     def decode(cls, auth_header: str, encoding: str='latin1') -> 'BasicAuth':
         """Create a BasicAuth object from an Authorization HTTP header."""
-        split = auth_header.strip().split(' ')
-        if len(split) == 2:
-            if split[0].strip().lower() != 'basic':
-                raise ValueError('Unknown authorization method %s' % split[0])
-            to_decode = split[1]
-        else:
+        try:
+            auth_type, encoded_credentials = auth_header.split(' ', 1)
+        except ValueError:
             raise ValueError('Could not parse authorization header.')
 
+        if auth_type.lower() != 'basic':
+            raise ValueError('Unknown authorization method %s' % auth_type)
+
         try:
-            username, _, password = base64.b64decode(
-                to_decode.encode('ascii')
-            ).decode(encoding).partition(':')
+            decoded = base64.b64decode(
+                encoded_credentials.encode('ascii'), validate=True
+            ).decode(encoding)
```

Comment:
Not sure whether it's applicable to current PR, but JFYI some browsers (Firefox) don't encode some of UTF-8 bytes correctly, assuming `ISO-8859-1` for unicode input and loosely encodes that input, which cuts some bytes in two-byte encoded chars (try entering `€öäü` in browser and see what you receive in server), which results in error during encoding since it cannot understand byte sequence when reaches those characters. Ref https://github.com/cherrypy/cherrypy/issues/1680

Reply:
Thanks for the reminder.
We discussed it 2 or 3 years ago and decided to do nothing until users report.
There is no blame yet :)

Reply:
thanks @webknjaz. noted, but it seems to be out of scope of this PR though.